lib/api.rb

```diff
@@ -69,13 +116,23 @@ def validation_error(msg)
 
   # we're always going to reply with json
   content_type :json
+  response.headers['Access-Control-Allow-Origin'] = '*'
 end
 
 namespace IDENTITY_BASE_URL do
   # depending on grant_type:
   #  password: login with a username/password, register/update the device
   #  refresh_token: just generate a new access_token
   # respond with an access_token and refresh_token
+  
+  # For HTTP CORS verification
+  options "*" do
+    response.headers["Allow"] = "GET, POST, OPTIONS, PUT, DELETE"
+    response.headers["Access-Control-Allow-Headers"] = "Authorization, Content-Type, Accept"
+    response.headers["Access-Control-Allow-Origin"] = "*"
+    200
+  end
```

Comment:
if the web-app is hosted under the same domain as the backend, then this is also not needed.

Reply:
You’re right. But it need a reverse proxy in front (that’s what i did).